src/entrypoints/background/context-menu.ts

```diff
@@ -0,0 +1,211 @@
+import type { Browser } from '#imports'
+import type { Config } from '@/types/config/config'
+import { browser, i18n, storage } from '#imports'
+import { isAPIProviderConfig } from '@/types/config/provider'
+import { getProviderConfigById } from '@/utils/config/helpers'
+import { CONFIG_STORAGE_KEY } from '@/utils/constants/config'
+import { getTranslationStateKey, TRANSLATION_STATE_KEY_PREFIX } from '@/utils/constants/storage-keys'
+import { logger } from '@/utils/logger'
+import { sendMessage } from '@/utils/message'
+
+const MENU_ID_TRANSLATE = 'read-frog-translate'
+
+let currentConfig: Config | null = null
+
+/**
+ * Initialize context menu based on config
+ */
+export async function setupContextMenu() {
+  const config = await storage.getItem<Config>(`local:${CONFIG_STORAGE_KEY}`)
+  if (!config) {
+    return
+  }
+
+  currentConfig = config
+  await updateContextMenuItems(config)
+
+  // Listen for config changes
+  storage.watch<Config>(`local:${CONFIG_STORAGE_KEY}`, async (newConfig) => {
```

Comment:
service worker might be inactive when you don't interact with Read Frog for minutes. then this watch might not able to always watch the change.

Reply:
Btw, you should manually test this even you AI tell you it fixed this. because this is very tricky and AI might be not able to understand it well.